### DISCUSSION TITLE
(nuxt3) is there a way to update the header data ?

### DISCUSSION THREAD

**Discussion by proalaa** [nuxt-community/apollo-module#481] [Q&A]
Hi Here 

I'm working on a nuxt 3 project and I use nuxt/apollo to consume backend graphql api

I need to set `authorization` and `cart-session-id` in header to be able to get data from server 

they both retrieved from server after login process

and I was able to set the `authorization` successfully through onLogin helper

but I find no way to set the `cart-session-id` 

your help is highly appreciated

**Comment by NathanD19** [comment]
Hey @proalaa, did you find a way to do this?

**Reply by proalaa** [reply]
finally I followed this soloution
https://github.com/nuxt-modules/apollo/issues/442#issuecomment-1420860945
